moment.parseFormat.js

```diff
@@ -136,6 +146,27 @@
     // Lets check for 4 digits first, these are years for sure
     format = format.replace(regexYearLong, 'YYYY')
 
+    // check if both numbers are < 13, then it must be DD/MM
+    format = format.replace(regexDayShortMonthShort, 'D/M')
+
+    // check if first number is < 10 && last < 13, then it must be D/MM
+    format = format.replace(regexDayShortMonth, 'D/MM')
+
+    // check if last number is < 32 && last < 10, then it must be DD/M
+    format = format.replace(regexDayMonthShort, 'DD/M')
+
+    // check if both numbers are > 10, but first < 32 && last < 13, then it must be DD/MM
+    format = format.replace(regexDayMonth, 'DD/MM')
+
+    // check if first < 10 && last > 12, then it must be M/YY
+    format = format.replace(regexMonthShortYearShort, 'M/YY')
+
+    // check if first < 13 && last > 12, then it must be MM/YY
+    format = format.replace(regexMonthYearShort, 'MM/YY')
+
+    // check if both numbers are < 12, then it must be something wrong
+    format = format.replace(regexDayYear, 'undefined')
```

Comment:
if both numbers are > 12, then it's not a valid date, in which case I wouldn't replace it at all